code/datums/quirks/negative.dm

```diff
@@ -60,17 +60,32 @@
 	medical_record_text = "Patient requires regular treatment for blood loss due to low production of blood."
 	hardcore_value = 8
 	processing_quirk = TRUE
+	var/bloodless_species = FALSE
+
+/datum/quirk/blooddeficiency/add()
```

Comment:
``add()`` is called before they check for ``processing_quirk``, I don't think you have to stop processing here at all.